node/src/components.rs

```diff
@@ -1,7 +1,49 @@
-//! Components
+//! Components subsystem.
 //!
-//! Components are the building blocks of the whole application, wired together inside a reactor.
-//! Each component has a unified interface, expressed by the `Component` trait.
+//! Components are the building blocks for the application and wired together inside a
+//! [reactor](crate::reactor). Each component has a unified interface, expressed by the
+//! [`Component`] trait.
+//!
+//! # Events
+//!
+//! Every component defines a set of events it can process, expressed through the
+//! [`Component::Event`] associated type. If an event that originated outside the component is to be
+//! handled (e.g. a request or announcement being handled), a `From<OutsideEvent> for
+//! ComponentEvent` implementation must be added (see component vs reactor event section below).
+//!
+//! A typical cycle for components is to receive an event, either originating from the outside, or
+//! as the result of an effect created by the component. This event is processed in the
+//! [`handle_event`](Component::handle_event) function, potentially returning effects that may
+//! produce new events.
+//!
+//! # Error and halting states
+//!
+//! Components in general are expected to be able to handle every input (that is every
+//! [`Component::Event`]) in every state. Unexpected inputs should usually be logged and discard, if
```

Comment:
```suggestion
//! [`Component::Event`]) in every state. Unexpected inputs should usually be logged and discarded, if
```

Reply:
Have to fix this manually due to the linebreak :)